### PR TITLE
Update the transfer learning notebook to Keras 3

### PR DESCRIPTION
**Pull request by takumiohym**
Update Summary:
- Renamed the file
- Updated to Keras 3
- Updated TF Hub to Keras Hub
- Simplified the data loading process with `keras.utils.image_dataset_from_directory` to focus on the main topic (and 10x faster)
- Removed the data augmentation since the method was too old. I'll create another notebook dedicated for data augmentation to cover Keras-native method and advanced data augmentation.
- Added Fine-tuning section
- Added Learning Rate Schedule section

Will clean the outputs and add lab version after the initial review.

### CODE REVIEW

**Review by sanjanalreddy — APPROVED** [review]

**Comment by takumiohym** [comment]
<p>Yes, the ouptuts were kept for review purpose as I mentioned in the PR comment. Now I'll remove them!</p>

---

 <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/658/discussion/#comment-3260228899'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by takumiohym** [comment]
<p>This was a placeholder for the new notebook name <code>2_mnist_cnn.ipynb</code> . Now it should work since the notebook was merged and this branch was rebased on that!</p>

---

 <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/658/discussion/#comment-3260228913'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by takumiohym** [comment]
<p>Added. Thanks!</p>

---

 <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/658/discussion/#comment-3260228921'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by takumiohym** [comment]
<p>Updated!</p>

---

 <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/658/discussion/#comment-3260228928'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by takumiohym** [comment]
<p>Updated!</p>

---

 <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/658/discussion/#comment-3260228944'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by takumiohym** [comment]
<p>Updated!</p>

---

 <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/658/discussion/#comment-3260228956'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>
